File: backend/src/main/java/com/spareparts/inventory/entity/PurchaseItem.java
package com.spareparts.inventory.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Entity
@Table(name = "purchase_items")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PurchaseItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "purchase_id", nullable = false)
    private Purchase purchase;

    @NotBlank
    @Column(name = "product_name")
    private String productName;

    @Column(name = "part_number")
    private String partNumber;

    @NotNull
    private Integer quantity;

    @NotNull
    @Column(name = "cost_price", precision = 10, scale = 2)
    private BigDecimal costPrice;

    @Column(name = "selling_price", precision = 10, scale = 2)
    private BigDecimal sellingPrice;

    @Column(name = "gst", precision = 10, scale = 2)
    private BigDecimal gst;

    @NotNull
    @Column(name = "total_amount", precision = 10, scale = 2)
    private BigDecimal totalAmount;
}

File: spare_parts_app_new/lib/models/purchase_item.dart
class PurchaseItem {
  final int? id;
  final String productName;
  final String? partNumber;
  final int quantity;
  final double costPrice;
  final double? sellingPrice;
  final double? gst;
  final double totalAmount;

  PurchaseItem({
    this.id,
    required this.productName,
    this.partNumber,
    required this.quantity,
    required this.costPrice,
    this.sellingPrice,
    this.gst,
    required this.totalAmount,
  });

  factory PurchaseItem.fromJson(Map<String, dynamic> json) {
    return PurchaseItem(
      id: json['id'],
      productName: json['productName'] ?? '',
      partNumber: json['partNumber'],
      quantity: json['quantity'] ?? 0,
      costPrice: (json['costPrice'] ?? 0).toDouble(),
      sellingPrice: json['sellingPrice']?.toDouble(),
      gst: json['gst']?.toDouble(),
      totalAmount: (json['totalAmount'] ?? 0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'productName': productName,
      'partNumber': partNumber,
      'quantity': quantity,
      'costPrice': costPrice,
      'sellingPrice': sellingPrice,
      'gst': gst,
      'totalAmount': totalAmount,
    };
  }
}
